src/uct/ib/ud/base/ud_ep.h

```diff
@@ -388,11 +390,20 @@ uct_ud_ep_ctl_op_check_ex(uct_ud_ep_t *ep, uint32_t ops)
            ((ep->tx.pending.ops & ~ops) == 0);
 }
 
-
-/* TODO: relay on window check instead. max_psn = psn  */
+/* TODO: rely on window check instead. max_psn = psn  */
 static UCS_F_ALWAYS_INLINE int uct_ud_ep_is_connected(uct_ud_ep_t *ep)
 {
-    return ep->dest_ep_id != UCT_UD_EP_NULL_ID;
+    ucs_assert((ep->dest_ep_id == UCT_UD_EP_NULL_ID) ||
+               (ep->flags & UCT_UD_EP_FLAG_CONNECTED));
```

Comment:
should it be 
ucs_assert((ep->dest_ep_id == UCT_UD_EP_NULL_ID) == !(ep->flags & UCT_UD_EP_FLAG_CONNECTED))